service-account/bound-sa-token.md

```diff
@@ -1,122 +0,0 @@
-
--> # Bound Service Account Token <-
-
---------------------------------------------------
--> # Table of Contents <-
-==============
-
--> # What is a Service Account Token  <-
--> # Service Account Token as a YAML <-
--> # Old Service Account Token Behavior  <-
--> # What is a Bound Service Account Token  <-
--> # Lord of Tokens: Kubelet <-
--> # Lord of ServiceAccounts: Service Account Admission Controller <-
--> # Bound Service Account Token in use by a Pod <-
-
--------------------------------------------------
--> # What is a Service Account Token  <-
-
-An **identity** that an instance or an application can use to make **API requests**:
-
-1. Intended to be used for **processes**.
-2. Bound to a **namespace**.
-3. Being associated with a ServiceAccount provides a **Bearer Token**.
-
--------------------------------------------------
--> # Service Account Token as a YAML <-
-
-```yaml
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: my-sa
-  namespace: my-namespace
-```
-
--------------------------------------------------
--> # Old Service Account Token Behavior  <-
-
-
-1. Previously **didn't expire**.
-2. Previously could be **used anywhere**.
-3. Token was **stored in a secret**.
-
--------------------------------------------------
--> # What is a Bound Service Account Token  <-
-
-- A Service Account Token that is **bound** to an **audience and lifespan**.
-- Given by **volume projection** or **TokenRequestAPI**.
-
--------------------------------------------------
--> # Lord of Tokens: Kubelet <-
-
-- Tries to **rotate** it after 80% of its lifetime is over.
-- Acquires it from the `apiserver` through the `TokenRequestAPI`.
-
--------------------------------------------------
--> # Lord of ServiceAccounts: Service Account Admission Controller <-
-
-In case that a **Pod Manifest** doesn't have a `ServiceAccount` specified:
-- adds the default `ServiceAccount`,
-- adds a **projected volume** and
-- adds `ImagePullSecrets`
-
-Runs within the `apiserver`.
-
--------------------------------------------------
--> # Bound Service Account Token in use by a Pod <-
-
-```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: nginx
-spec:
-  containers:
-  - image: nginx
-    name: nginx
-    volumeMounts:
-    - mountPath: /var/run/secrets/tokens
-      name: vault-token
-  serviceAccountName: build-robot 
-  volumes:
-  - name: vault-token
-    projected:
-      sources:
-      - serviceAccountToken:
-          path: vault-token 
-          expirationSeconds: 7200 
-          audience: vault
-```
-
--------------------------------------------------
--> # Legacy Usage <-
-
-Creates a secret with the token inside:
-
-```yaml
-apiVersion: v1
-kind: Secret
-metadata:
-  name: build-robot-secret
-  annotations:
-    kubernetes.io/service-account.name: build-robot
-type: kubernetes.io/service-account-token
-```
-
--------------------------------------------------
--> # Lord of Tokens: Token Controller <-
-
-- Creates `ServiceAccount` `secret`
```

Comment:
Did you mean `Creates a secret for a service account token` ?

Questions that the audience shouldn't need to ask:
- is this the legacy behavior?
- is this on by default?